# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/MediaTrackGraph.cpp
# Commit: a00ad7266f54
# Full Hash: a00ad7266f54820d7a3ff1233fa156f63190a3a3
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2020-11-06 04:14:34
# Regressor Bug: 1651745
# File Overlap Count: 1
# Description:
#   Bug 1651745 - Signal AudioInputProcessing on input stream stop instead of driver start. r=padenot
#   
#   Without this patch, AudioInputProcessing wouldn't be aware of an audio driver
#   changing to another driver, be it a system driver or an audio driver that starts
#   on its fallback.
# ==============================================================================

diff -r a76d9158cdfd -r a00ad7266f54 dom/media/MediaTrackGraph.cpp
--- a/dom/media/MediaTrackGraph.cpp	Thu Nov 05 16:42:41 2020 +0000
+++ b/dom/media/MediaTrackGraph.cpp	Thu Nov 05 16:43:24 2020 +0000
@@ -816,7 +816,7 @@
   }
 }
 
-void MediaTrackGraphImpl::NotifyStarted() {
+void MediaTrackGraphImpl::NotifyInputStopped() {
 #ifdef ANDROID
   if (!mInputDeviceUsers.GetValue(mInputDeviceID)) {
     return;
@@ -830,7 +830,7 @@
       mInputDeviceUsers.GetValue(mInputDeviceID);
   MOZ_ASSERT(listeners);
   for (auto& listener : *listeners) {
-    listener->NotifyStarted(this);
+    listener->NotifyInputStopped(this);
   }
 }
 